frontend/sozisel-app/src/components/SessionResultScreen/SessionResultEvents/EventResultDetails/Modules/QuizResultDetails/QuizResultViews/Participants/QuizResultParticipantsView.tsx

```diff
@@ -0,0 +1,165 @@
+import { ensure } from "../../../../../../../utils/Typescript/ensure";
+import { useTranslation } from "react-i18next";
+
+export interface QuizResultParticpantsViewProps {
+  id: string;
+}
+export default function QuizResultParticipantsView({
+  id,
+}: QuizResultParticpantsViewProps): React.ReactElement {
+  const { t } = useTranslation("common");
+  const [currentParticipant, setCurrentParticipant] =
+    useState<QuizParticipantSummary | null>(null);
+  const { loading, data } = useQuizParticipantsSummaryQuery({
+    variables: { id },
+  });
+  const { loading: questionsAndAnswersLoading, data: questionsAndAnswers } =
+    useQuizQuestionsAndAnswersQuery({ variables: { id } });
+
+  const getParticipantStats = () => {
+    return [
+      {
+        label: t("components.SessionEventResults.Quiz.points"),
+        value: `${currentParticipant?.numberOfPoints.toFixed(2)}`,
+      },
+      {
+        label: t("components.SessionEventResults.Quiz.answerTime"),
+        value: `${currentParticipant?.quizAnswerTime.toFixed(2)}`,
+      },
+    ];
+  };
+
+  if (loading || questionsAndAnswersLoading) {
+    return (
+      <div className="QuizResultParticipantsView">
+        <CircularProgress />
+      </div>
+    );
+  }
+
+  if (
+    data?.quizParticipantsSummary &&
+    questionsAndAnswers?.quizQuestionsSummary
+  ) {
+    return (
+      <>
+        <div className="QuizResultParticipantsView">
+          <div className="headerWithIcon">
+            <PeopleIcon color="primary" fontSize="large" />
+            <Typography variant="h3" className="header">
+              {t("components.SessionEventResults.Quiz.participants")}
+            </Typography>
+          </div>
+          <EnhancedTable
+            data={data.quizParticipantsSummary}
+            headCells={[
+              {
+                id: "fullName",
+                label: t("components.SessionEventResults.Quiz.name"),
+              },
+              {
+                id: "numberOfPoints",
+                label: t("components.SessionEventResults.Quiz.points"),
+              },
+              {
+                id: "quizAnswerTime",
+                label: t("components.SessionEventResults.Quiz.answerTime"),
+              },
+            ]}
+            onClick={(participant) => {
+              setCurrentParticipant(participant);
+            }}
+          />
+        </div>
+
+        <QuizResultDetailsDialog
+          isOpen={currentParticipant !== null}
+          handleClose={() => setCurrentParticipant(null)}
+          title={t("components.SessionEventResults.Quiz.participantDetails")}
+          detailName={currentParticipant?.fullName ?? ""}
+          detailIcon={<PersonIcon color="primary" fontSize="large" />}
+          stats={getParticipantStats()}
+          detailsViewTitle={t(
+            "components.SessionEventResults.Quiz.resultForQuestions"
+          )}
+          chartData={(currentParticipant?.participantAnswers ?? []).map(
```

Comment:
All the code starting from here down needs a refactor. Don't you feel guilty when typing that kind of monstrosity?